### PR TITLE
Start a new rewarding cycle + add current cycle page

### PR DESCRIPTION
**Pull request by RazvanN7**
cc @CyberShadow

### CODE REVIEW

File: source/dlangbot/app.d [hunk 1]
```diff
@@ -457,13 +458,31 @@ void agentShutdownCheck(HTTPServerRequest req, HTTPServerResponse res)
...
 
-void contributorStats(HTTPServerRequest req, HTTPServerResponse res)
+void contributorStatsOverall(HTTPServerRequest req, HTTPServerResponse res)
+{
+    contributorStats(req, res);
+}
+
+void contributorStatsPerCycle(HTTPServerRequest req, HTTPServerResponse res)
+{
+    contributorStats(req, res, "2022-01-01 00:00:00.000000", "2022-03-01 00:00:00.000000");
```
Comment (CyberShadow) [inline source/dlangbot/app.d:468]: What do you think about using some date arithmetics to automatically get the current cycle?

File: source/dlangbot/app.d [hunk 2]
```diff
@@ -50,7 +50,8 @@ void startServer(HTTPServerSettings settings)
         .post("/codecov_hook", &codecovHook)
         .post("/buildkite_hook", &buildkiteHook)
         .post("/agent_shutdown_check", &agentShutdownCheck)
-        .get("/contributor_stats", &contributorStats)
```
Comment (CyberShadow) [inline source/dlangbot/app.d:53]: Note, this will break existing links and will require changes on dlang.org.

**Review by CyberShadow — APPROVED** [review]

File: source/dlangbot/app.d [hunk 3]
```diff
@@ -457,13 +458,31 @@ void agentShutdownCheck(HTTPServerRequest req, HTTPServerResponse res)
...
+private void contributorStats(HTTPServerRequest req, HTTPServerResponse res,
+        string startDate = "2021-01-01 00:00:00.000000", string endDate = "2100-01-01 00:00:00.000000")
 {
     import dlangbot.database : getContributorsStats;
     import vibe.http.server : render;
 
-    string[][] entries = getContributorsStats("2021-01-01 00:00:00.000000", "2100-01-01 00:00:00.000000");
-    res.render!("contributor_stats.dt", entries);
+    string[][] entries = getContributorsStats(startDate, endDate);
+    if (startDate = "2021-01-01 00:00:00.000000")
```
Comment (CyberShadow) [inline source/dlangbot/app.d:478]: Should be `==`

File: source/dlangbot/app.d (hunk 2 above)
Comment (RazvanN7) [reply source/dlangbot/app.d:53]: Ok, I can leave the name unchanged.

File: source/dlangbot/app.d (hunk 1 above)
Comment (RazvanN7) [reply source/dlangbot/app.d:468]: I don't know for sure how these cycles will change so I prefer to hard code them for now.

**Comment by RazvanN7** [comment]
@CyberShadow I think this is ready now.

**Pull request merged**